### DISCUSSION TITLE
post_233924

### DISCUSSION THREAD

**Discussion by giscus** [cincuranet/blog-comments#1355] [Comments]
# post_233924

https://www.tabsoverspaces.com/233924-csharp-class-named-end

**Comment by Rand-Random** [comment]
Maybe its my lack of creativity, but why exactly is a class called "end" dumb?
as a counter example is start a non-dumb name?

**Reply by cincuranet** [reply]
First of all, it violates general naming conventions in C#. Second of all, `end` (or `start`, or `begin`) is so generic keyword that sooner or later you'll have collision somewhere and it will create unnecessary work to do escaping, quoting, ...

**Comment by rlundy** [comment]
Have fun while it lasts, because it won't last long:

https://blog.paranoidcoding.org/2022/04/11/lowercase-type-names.html

As the article mentions, the whole reason we use async/await in C# instead of just await is because "await" could be a type name, so the async is necessary to avoid ambiguity.  And having to include this "marker word" async is a common developer complaint, so Microsoft is moving toward ending allowing lower-cased type names so as to avoid this kind of extraneous syntax in the future.

**Reply by cincuranet** [reply]
Oh yeah, I'm well aware of [this from the warning waves](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/warning-waves#cs8981---the-type-name-only-contains-lower-cased-ascii-characters). But warnings are for weak. ;)
